pkg/serverless/appsec/config.go

```diff
@@ -0,0 +1,151 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016 Datadog, Inc.
+
+package appsec
+
+import (
+	"fmt"
+	"os"
+	"regexp"
+	"strconv"
+	"time"
+	"unicode"
+	"unicode/utf8"
+
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+)
+
+const (
+	enabledEnvVar         = "DD_APPSEC_ENABLED"
+	rulesEnvVar           = "DD_APPSEC_RULES"
+	wafTimeoutEnvVar      = "DD_APPSEC_WAF_TIMEOUT"
+	obfuscatorKeyEnvVar   = "DD_APPSEC_OBFUSCATION_PARAMETER_KEY_REGEXP"
+	obfuscatorValueEnvVar = "DD_APPSEC_OBFUSCATION_PARAMETER_VALUE_REGEXP"
+)
+
+const (
+	defaultWAFTimeout           = 4 * time.Millisecond
+	defaultObfuscatorKeyRegex   = `(?i)(?:p(?:ass)?w(?:or)?d|pass(?:_?phrase)?|secret|(?:api_?|private_?|public_?)key)|token|consumer_?(?:id|key|secret)|sign(?:ed|ature)|bearer|authorization`
+	defaultObfuscatorValueRegex = `(?i)(?:p(?:ass)?w(?:or)?d|pass(?:_?phrase)?|secret|(?:api_?|private_?|public_?|access_?|secret_?)key(?:_?id)?|token|consumer_?(?:id|key|secret)|sign(?:ed|ature)?|auth(?:entication|orization)?)(?:\s*=[^;]|"\s*:\s*"[^"]+")|bearer\s+[a-z0-9\._\-]+|token:[a-z0-9]{13}|gh[opsu]_[0-9a-zA-Z]{36}|ey[I-L][\w=-]+\.ey[I-L][\w=-]+(?:\.[\w.+\/=-]+)?|[\-]{5}BEGIN[a-z\s]+PRIVATE\sKEY[\-]{5}[^\-]+[\-]{5}END[a-z\s]+PRIVATE\sKEY|ssh-rsa\s*[a-z0-9\/\.+]{100,}`
+)
+
+// StartOption is used to customize the AppSec configuration when invoked with appsec.Start()
+type StartOption func(c *Config)
+
+// Config is the AppSec configuration.
+type Config struct {
+	// rules loaded via the env var DD_APPSEC_RULES. When not set, the builtin rules will be used.
+	rules []byte
+	// Maximum WAF execution time
+	wafTimeout time.Duration
+	// AppSec trace rate limit (traces per second).
+	traceRateLimit uint
+	// Obfuscator configuration parameters
+	obfuscator ObfuscatorConfig
+}
+
+// ObfuscatorConfig wraps the key and value regexp to be passed to the WAF to perform obfuscation.
+type ObfuscatorConfig struct {
+	KeyRegex   string
+	ValueRegex string
+}
+
+// isEnabled returns true when appsec is enabled when the environment variable
+// It also returns whether the env var is actually set in the env or not
+// DD_APPSEC_ENABLED is set to true.
+func isEnabled() (enabled bool, set bool, err error) {
+	enabledStr, set := os.LookupEnv(enabledEnvVar)
```

Comment:
You can simplify a lot this function with helpers from the config package such as `config.Datadog.GetBool`

Reply:
Postponed to public beta.